File: framework-ts/module/devian-domain-game/features/ability/AbilityBattleStatus.ts
import { STATUS } from '../../Generated/Game.g';
import { AbilityBattleBase } from './AbilityBattleBase';

export class AbilityBattleStatus extends AbilityBattleBase {
    private mTable: STATUS | null = null;

    get statusId(): string { return this.mTable?.status_id ?? ''; }
    get nameId(): string { return this.mTable?.name_id ?? ''; }
    get affectList(): readonly string[] { return this.mTable?.affect_list ?? []; }

    init(table: STATUS): void {
        this.mTable = table;
    }

    clone(): AbilityBattleStatus {
        const c = new AbilityBattleStatus();
        c.mTable = this.mTable;
        c.copyStatsFrom(this);
        return c;
    }
}

File: framework-cs/apps/UnityExample/Assets/Samples/Devian Foundation/0.1.0/GamePackage/Runtime/Ability/AbilityBattleStatus.cs
using System;
using System.Collections.Generic;
using Devian.Domain.Game;

namespace Devian
{
    public sealed class AbilityBattleStatus : AbilityBattleBase
    {
        STATUS mTable = null;

        public string StatusId => mTable?.status_id ?? string.Empty;
        public string NameId => mTable?.name_id ?? string.Empty;
        public IReadOnlyList<string> AffectList => mTable != null ? mTable.affect_list : Array.Empty<string>();

        public void Init(STATUS table)
        {
            mTable = table;
        }

        public override AbilityBase Clone()
        {
            var c = new AbilityBattleStatus();
            c.mTable = mTable;
            c.CopyStatsFrom(this);
            return c;
        }
    }
}
